js/Puzzle.js

```diff
@@ -3,6 +3,9 @@ class Puzzle {
     this.currentPuzzle = this.generatePuzzle(round);
     this.puzzleCompleted = false;
   }
+
+  // keep track of revealed letters already guessed in current puzzle
```

Comment:
This is a good point. We may have to put already guessed letters into an array of guessed letters.